### ISSUE TITLE
Parsing error: Expecting Unicode escape sequence \uXXXX

### ISSUE THREAD

**Issue by EgidioCaprino** [eslint/eslint#8952]
**Tell us about your environment**
* **ESLint Version:** 3.19.0

* **Node Version:** 8.1.4
* **npm Version:** 5.3.0

**What parser (default, Babel-ESLint, etc.) are you using?** Babel-ESLint

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "extends": "airbnb",
  "env": {
    "mocha": true,
    "browser": true
  },
  "rules": {
    "react/jsx-filename-extension": [
      1,
      {
        "extensions": [
          ".js",
          ".jsx"
        ]
      }
    ],
    "react/jsx-no-bind": [
      1,
      {
        "allowBind": true
      }
    ]
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue.**
Try to `eslint` the following file:

```js
const defaultServices = {
  lastTableId: 0,
  tables: [],
  getNextTableId: () => { this.lastTableId += 1; return this.lastTableId; },
  delay: 3000,
};

export const saveTable = (
  table,
  { tables, delay, getNextTableId } = defaultServices,
) => new Promise((resolve, reject) => {
  setTimeout(() => {
    const nameExists = tables.some(({ name }) => name.toLowerCase() === table.name.toLowerCase());
    if (nameExists) {
      reject(new Error(`Table with name '${table.name}' already exists.`));
    } else {
      const savedTable = {
        ...table,
        id: getNextTableId(),
        qrCodeImageUrl: 'http://www.appcoda.com/wp-content/uploads/2013/12/qrcode.jpg',
      };
      tables.push(savedTable);
      resolve(savedTable);
    }
  }, delay);
});

export const getTables = ({ tables, delay } = defaultServices) => new Promise((resolve) => {
  setTimeout(() => {
    resolve([...tables]);
  }, delay);
});

export const deleteTable = (
  tableId,
  { tables, delay } = defaultServices,
) => new Promise((resolve, reject) => {
  setTimeout(() => {
    const index = tables.findIndex(({ id }) => id === tableId);
    if (index < 0) {
      reject(new Error(`Table with ID ${tableId} not found.`));
    } else {
      const [deletedTable] = tables.splice(index, 1);
      resolve(deletedTable);
    }
  }, delay);
});
```

**What did you expect to happen?**
To pass the lint or to show useful messages.

**What actually happened? Please include the actual, raw output from ESLint.**

> beat-the-waiter-client@1.0.0 eslint /home/egidio/Documents/beat-the-waiter-client
> eslint src test --fix

/home/egidio/Documents/beat-the-waiter-client/test/common/mock-api.js
  8:32  error  Parsing error: Expecting Unicode escape sequence \uXXXX

✖ 1 problem (1 error, 0 warnings)

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! beat-the-waiter-client@1.0.0 eslint: `eslint src test --fix`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the beat-the-waiter-client@1.0.0 eslint script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/egidio/.npm/_logs/2017-07-15T17_33_18_507Z-debug.log
